Add Authoritative Zone resource

bsmithtm opened the pull request:
Fixes #237.

This adds a new `infoblox_zone_auth` resource with the functions required to create and manage them. It includes a new doc for the new resource with example usage as well.

NOTE: It depends on the following PR being merged in the infoblox-go-client, to expose the necessary Authoritative Zone fields.

* PR: https://github.com/infobloxopen/infoblox-go-client/pull/186

skudriavtsev commented on infoblox/resource_infoblox_zone_auth.go:
```diff
@@ -0,0 +1,179 @@
+				Optional:    true,
+				Description: "Extensible attributes, as a map in JSON format",
+			},
+		},
```
There must be the field about DNS view, and preferably about zone format (forward or reverse, like in WAPI).

bsmithtm replied:
I added these in the matching Go Client PR above and to the resource here in the provider

skudriavtsev commented on infoblox/resource_infoblox_zone_auth.go:
```diff
@@ -0,0 +1,179 @@
+}
+
+func resourceZoneAuthUpdate(d *schema.ResourceData, m interface{}) error {
+	return nil
```
This is a bad way for not supporting updates for a resource. Please, change this, either by writing the actual code for 'UPDATE' operation or by clearly indicating that the resource update is not supported so far.

bsmithtm replied:
I agree; I've created the Update function in the Go Client and used it here. More details in the comment below.

bsmithtm commented:
Hi @skudriavtsev thanks again for the review and comments!

The Update function had not been implemented in the Go Client when I originally created this PR.

Over in the matching Go Client PR I have now implemented the `UpdateZoneAuth()` function. Here I've implemented `resourceZoneAuthUpdate()` to use the new function and succeeded with our internal testing.

The fields that cannot be updated in the WAPI (`fqdn` and `zone_format`) are left out of the Update function; all other fields are updatable with this addition.

Thanks!

JkhatriInfobox commented:
This is already supported from v2.5.0 onwards.

Hence closing this PR
